aiobotocore/credentials.py

```diff
@@ -376,11 +376,6 @@ def __init__(self, refresh_using, method, time_fetcher=_local_now):
         self.method = method
         self._frozen_credentials = None
 
-    def refresh_needed(self, refresh_in=None):
-        if self._frozen_credentials is None:
-            return True
-        return super().refresh_needed(refresh_in)
-
```

Comment:
this method is still in botocore: https://github.com/boto/botocore/blob/1941b2b67bd033957c5ddd4855868fe7e010a174/botocore/credentials.py#L686-L689

Reply:
Yes, but since we fix the class hierarchy, we no longer need to patch it. We simply use upstream's implementation. It's non-blocking anyway.

Reply:
oo, wait, why is this not async, hmm

Reply:
ah ok, doesn't need to be async, yay, ignore!